crates/wasmtime/src/runtime/vm/cow.rs

```diff
@@ -610,26 +651,30 @@ impl MemoryImageSlot {
                     // Note that the memset may be zero bytes here.
 
                     // This is memset (1)
-                    ptr::write_bytes(self.base.as_ptr(), 0u8, keep_resident);
+                    ptr::write_bytes(self.base.as_ptr(), 0u8, keep_resident.byte_count());
 
                     // This is madvise (2)
                     self.restore_original_mapping(
                         keep_resident,
-                        self.accessible - keep_resident,
+                        self.accessible
+                            .checked_sub(keep_resident)
+                            .expect("XXX what should happen if this underflows?"),
```

Comment:
Wasn't clear to me whether this was possible -- there are several offsets here that all interact with each other.

Reply:
Okay -- to be honest I'm not entirely sure that this is the case in reality, it seems like in configuration you could define keep_resident to be more then the byte size. But in any case this should panic (and if there is such a condition it should be handled at a higher level)